File: Go/console/position/position.go
package position

import "errors"

type Position struct {
	Row    int
	Column int
}

func NewPosition(row int, column int) Position {
	if !IsValidPosition(row, column) {
		panic(errors.New("Invalid position index."))
	}

	return Position{Row: row, Column: column}
}

func IsValidPosition(row int, column int) bool {
	return isValidIndex(row) && isValidIndex(column)
}

func isValidIndex(index int) bool {
	return index >= 0 && index < 9
}

File: CSharp/Sudoku.Core/Models/Position.cs
namespace Sudoku.Models;

public record Position
{
	public int Row { get; }
	public int Column { get; }

    public Position(int row, int column)
	{
		if (!Table.IsValidIndex(row) || !Table.IsValidIndex(column))
		{
			throw new Exception($"Invalid position (r:{row},c:{column}");
		}

		Row = row;
		Column = column;
	}

    public override string ToString()
    {
		return $"(r:{Row},c:{Column})";

	}
}